tests/integration/test_increment_graph_window.py

```diff
@@ -0,0 +1,308 @@
+        window = {
+            'a': 0,
+            'b': 1,
+            'c': 2,
+            'd': 3,
+        }
+        # the tasks we will run in order
+        letters = 'abcdefghijklmnopqrs'
+        # the graph-depth of the "blink" task at each stage of the workflow
+        blink_distances = [1] + (list((*range(2, 5), *range(3, 0, -1))) * 3)
+
+        for ind, blink_distance in zip(range(len(letters)), blink_distances):
+            previous_task = letters[ind - 1] if ind > 0 else None
+            active_task = letters[ind]
+            yield (
+                previous_task,
+                active_task,
+                {
+                    # the tasks traversed via the "blink" task
+                    **blink[blink_distance],
+                    # the tasks in the main "abcdefg" chain
+                    **{key: abs(value) for key, value in window.items()},
+                }
+            )
+
+            # move each task in the "abcdef" chain down one
+            window = {key: value - 1 for key, value in window.items()}
+            # add the n=3 task in the "abcdef" chain into the window
+            with suppress(IndexError):
+                window[letters[ind + 4]] = 3
+            # pull out anything which is not supposed to be in the n=3 window
+            window = {
+                key: value
+                for key, value in window.items()
+                if abs(value) < 4
+            }
+
+    async with start(schd):
+        schd.data_store_mgr.set_graph_window_extent(3)
+        await schd.update_data_structure()
+
+        previous_n_window = {}
+        for previous_task, active_task, n_window in advance():
+            # mark the previous task as completed
+            await complete_task(schd, previous_task)
+            # add the next task to the pool
+            add_task(schd, active_task)
+            # run the graph window algorithm
+            increment_graph_window(schd, active_task)
+            # get the deltas which increment_graph_window created
+            added, updated, pruned = get_deltas(schd)
+
+            # compare the n-window in the store to what we were expecting
+            _n_window = await get_n_window(schd)
+            assert _n_window == n_window
+
+            # compate the deltas to what we were expecting
+            if active_task != 'a':
+                # skip the first task as this is complicated by startup logic
+                assert added == {
+                    key: value
+                    for key, value in _n_window.items()
+                    if key not in previous_n_window
+                }
+                # TODO: updated deltas don't seem to be coming through as expected
+                # assert {
+                #     key: value
+                #     for key, value in _n_window.items()
+                #     if previous_n_window.get(key, value) != value
+                # } == updated
```

Comment:
Any idea why the updated deltas are off?

Reply:
Will have a look

Reply:
Any ideas? It looks on the surface like the updated deltas are completely wrong, but I'm not sure whether that's genuine or whether I'm just accessing them wrong.

Reply:
fixed, needed to call `window_depth_finder`